src/pages/auth/login.tsx

```diff
@@ -1,40 +1,123 @@
+import authStyles from "../../styles/pages/auth/Auth.module.css";
 import {
   GetServerSideProps,
   InferGetServerSidePropsType,
   NextPage,
 } from "next";
 import { getCsrfToken } from "next-auth/react";
-import Link from "next/link";
 import { useRouter } from "next/router";
-
 import { getServerAuthSession } from "@server/common/getServerAuthSession";
+import {
+  Button,
+  ButtonGroup,
+  Card,
+  Elevation,
+  FormGroup,
+  Icon,
+  InputGroup,
+} from "@blueprintjs/core";
+import { ChangeEvent, useState } from "react";
+import { useDispatch } from "react-redux";
+import AuthTypes from "../../redux/types/authTypes";
+
+interface ILoginFormValues {
+  csrfToken: "";
+  email: string;
+  password: string;
+}
+
+const initialLoginFormValues: ILoginFormValues = {
+  csrfToken: "",
+  email: "",
+  password: "",
+};
 
 const Login: NextPage = ({
   csrfToken,
 }: InferGetServerSidePropsType<typeof getServerSideProps>) => {
+  const [formValues, setFormValues] = useState(initialLoginFormValues);
+
+  const dispatch = useDispatch();
   const router = useRouter();
+
+  const handleSignUpButtonClick = () => {
+    const href = {
+      pathname: "/auth/register",
+      query: { callbackUrl: router.query.callbackUrl },
+    };
+    router.push(href);
+  };
+
+  const handleLoginButtonClick = (): void => {
+    // TODO: validate inputs
+    setFormValues({ ...formValues, csrfToken });
+    dispatch({ type: AuthTypes.CREATE_LOGIN, payload: formValues });
+  };
+
+  const handleInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormValues({
+      ...formValues,
+      [name]: value,
+    });
+  };
+
   return (
-    <div>
-      <form method="post" action="/api/auth/callback/credentials">
-        <input name="csrfToken" type="hidden" defaultValue={csrfToken} />
-        <label>
-          Email
-          <input name="email" type="email" />
-        </label>
-        <label>
-          Password
-          <input name="password" type="password" />
-        </label>
-        <button type="submit">Sign in</button>
-      </form>
-      <Link
-        href={{
-          pathname: "/auth/register",
-          query: { callbackUrl: router.query.callbackUrl },
-        }}
+    <div className={authStyles.authContainer}>
+      <Card
+        className={authStyles.authFormCard}
+        interactive={false}
+        elevation={Elevation.THREE}
       >
-        <button>Create Account</button>
-      </Link>
+        <div className={authStyles.authFormCardHeader}>
+          <Icon icon="user" size={80} />
+          <h1>Login</h1>
+        </div>
+        <div className={authStyles.authForm}>
+          <FormGroup label="Email" labelFor="text-input" labelInfo="(Required)">
+            <InputGroup
+              id="text-input"
```

Comment:
This shares the same id as the password input. Chrome is giving a warning about it.
```
[DOM] Found 2 elements with non-unique id #text-input: (More info: https://goo.gl/9p2vKq)
<input type="password" id="text-input" placeholder="Password" class="bp4-input" name="password" value="a">
<input type="text" id="text-input" placeholder="Email" class="bp4-input" name="email" value="arkinmodi@gmail.com">
```